### PR TITLE
Py3: Fix vault tests

### PR DESCRIPTION
**Pull request by tiran**
* Bump PKI to 10.5.1-2, which fixes an issue with KRA under Python 3
* Correct encoding of secret, certs, keys, and salt

https://pagure.io/freeipa/issue/7033
Obsoletes: #1261
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: freeipa.spec.in [hunk 1]
```diff
@@ -192,7 +192,7 @@ BuildRequires:  python2-dns >= 1.15
 BuildRequires:  jsl
 BuildRequires:  python2-yubico
 # pki Python package
-BuildRequires:  pki-base-python2
+BuildRequires:  pki-base-python2 >= 10.5.1-2
```
Comment (rcritten) [inline freeipa.spec.in]: Do you know of plans to push this build to F27/rawhide?

Comment (tiran) [reply freeipa.spec.in]: No, somebody has to talk to Matt. @mharmsen99?

Comment (rcritten) [reply freeipa.spec.in]: Matt told us probably next week so that's fine. ACK.

**Comment by tiran** [comment]
master:

* 64a88d597cf2d65bc5f97a8fd399751acd5bebda Py3: Fix vault tests

**Pull request closed without merging**